pages/_app.tsx

```diff
@@ -1,6 +1,44 @@
-import '../styles/globals.css'
-import type { AppProps } from 'next/app'
+import "../styles/globals.css";
+import type { AppProps } from "next/app";
+import { useEffect, useState } from "react";
+import { Criteria, Member } from "../interfaces/interfaces";
 
 export default function App({ Component, pageProps }: AppProps) {
-  return <Component {...pageProps} />
+  const [desiredPlayer, setDesiredPlayer] = useState<Criteria>({
+    skill: "Anything will do",
+    language: "Anything will do",
+    gender: "Anything will do",
+  });
```

Comment:
These should be "empty" values I guess. Usually it's better to initialize them with `null` and then render "Anything will do" for the case they are `null`.

Reply:
this is an option for the filter of the array with all the players. I like it better if the default state of it is "anything will do" because all the players are rendered in the beginning. You can choose then another option, if you want to see only players with skill level A or anything else. I don´t understand why having a default value of null is an improvement of the readability of the code 🧐

Reply:
It‘s not about readability but about separation of concerns. Having no value set is something your data should indicate clearly, regardless of the actual display value. Imagine for example that your UI should be available in multiple languages, how would you then test if a value is set? You should always clearly separate display value and data.

Reply:
I think, there is a misunderstanding of the code. There is no no value set, I have a valid value from the beginning. When I create my data, there is a default setting of anything will do, that is a valid setting option. I change the value of the data through a on change attribute of a select element. That means, if the first option of the select is the desired one, there will be no change in the select element and no change in the data. This is why a create a data with a default value of the value of the first rendered option in the select element.

Reply:
Ok - I‘m still not fully convinced, but go for it :-)

Reply:
Surely there are more elegant ways to lose this problem with the default option of the select, but I need to have a working code and go on with other features now. I can refactor the code later when I have the time. 
I really appreciate your help, you made my code way better with this review. Thank you and have a nice weekend 🙂

Reply:
It‘s fine - just to clarify the point:
Later in your code, you will probably make some decisions based on the value of these variables, e.g.:

`if (skill===„Anything will do“){doSomething()} else {doSomethingDifferent()}`

From that point on it will become difficult to change the text because it‘s used as part of your logic.